app/forms/AssociateUcr.scala

```diff
@@ -48,8 +48,20 @@ object AssociateUcr {
 
     Forms.mapping(
       "kind" -> of[UcrType](UcrType.formatter),
-      "ducr" -> mandatoryIfEqual("kind", Ducr.formValue, text().verifying("ducr.error.format", validDucrIgnoreCase)),
-      "mucr" -> mandatoryIfEqual("kind", Mucr.formValue, text().verifying("mucr.error.format", validMucrIgnoreCase))
+      "ducr" -> mandatoryIfEqual(
+        "kind",
+        Ducr.formValue,
+        text()
+          .verifying("associate.ucr.ducr.error.empty", nonEmpty)
+          .verifying("associate.ucr.ducr.error.invalid", isEmpty or validDucrIgnoreCase)
```

Comment:
Do you need to double check if empty?

Reply:
They are not checking empty twice, the first verifying checks for if empty, the second verifying checks value is either empty or a valid value. This just prevents empty values from triggering the display of both empty and invalid error messages at the same time.